enforcer/enforcer_main.py

```diff
@@ -0,0 +1,249 @@
+import sys
+import os
+
+# Add parent directory to Python path so we can import enforcer modules  
+sys.path.insert(0, os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
+
+from enforcer.utils import add_custom_certificate
+
+ADDITIONAL_CERTIFICATE: str = os.environ.get("CERTIFICATE", "")
+
+if add_custom_certificate(ADDITIONAL_CERTIFICATE):
+    print("added custom certificate")
+
+# DO NOT ADD ANY CODE ABOVE THIS
+# ADDING IMPORTS BEFORE ADDING THE CUSTOM CERTS MIGHT INIT HTTP CLIENTS THAT DOESN'T RESPECT THE CUSTOM CERT
+
+import logging
+import os
+import sys
+import base64
+import json
+import re
+import time
+from typing import Dict, Any
+from fastapi import FastAPI, HTTPException, Response
+from pydantic import BaseModel
+from prometheus_client import generate_latest, CONTENT_TYPE_LATEST
+
+from enforcer.dal.supabase_dal import SupabaseDal
+from enforcer.patch_manager import patch_container_resources
+from enforcer.model import PodOwner, WorkloadRecommendation
+from enforcer.env_vars import ENFORCER_SSL_KEY_FILE, ENFORCER_SSL_CERT_FILE, KRR_MUTATION_MODE_DEFAULT
+from enforcer.resources.owner_store import OwnerStore
+from enforcer.resources.recommendation_store import RecommendationStore
+from enforcer.metrics import pod_admission_mutations, replicaset_admissions, rs_owners_size, admission_duration
+
+# Configure logging
+logger = logging.getLogger()
+logHandler = logging.StreamHandler(sys.stdout)
+formatter = logging.Formatter('%(asctime)s - %(levelname)s - %(message)s')
+logHandler.setFormatter(formatter)
+logger.addHandler(logHandler)
+logger.setLevel(os.environ.get("LOG_LEVEL", "INFO"))
+
+# Define the mention pattern regex
+MENTION_PATTERN = re.compile(r'@[\w.-]+')
+ENFORCE = "enforce"
+IGNORE = "ignore"
+
+app = FastAPI(
+    title="KRR Enforcer mutation webhook",
+    description="A KRR recommendations mutating webhook server for Kubernetes",
+    version="1.0.0"
+)
+
+dal = SupabaseDal()
+recommendation_store = RecommendationStore(dal)
+owner_store = OwnerStore()
+
+class AdmissionReview(BaseModel):
+    apiVersion: str
+    kind: str
+    request: Dict[str, Any]
+
+def admission_allowed(request: AdmissionReview) -> Dict[str, Any]:
+    return \
+        {
+            "apiVersion": "admission.k8s.io/v1",
```

Comment:
I believe in the admissionReviewVersions you're also allowing v1beta1. For v1vbeta1 is it ok to return v1 here?

Reply:
removing the support for v1beta1